R/bambu-extendAnnotations-utilityExtend.R

```diff
@@ -80,25 +84,36 @@ filterTranscriptsByAnnotation <- function(rowDataCombined, annotationGrangesList
   
   #calculate relative read count before any filtering
   rowDataCombined <- group_by(rowDataCombined, GENEID) %>% mutate(relReadCount = readCount/sum(readCount))
+  
+  rowDataCombined$TXNAME[is.na(rowDataCombined$TXNAME)] <- paste0(
+      prefix, "Tx", seq_len(sum(is.na(rowDataCombined$TXNAME))))
+  names(exonRangesCombined) <- rowDataCombined$TXNAME
+
+  #filter out subset reads
   if (remove.subsetTx) { # (1) based on compatibility with annotations
-    notCompatibleIds <- which(!grepl("compatible", rowDataCombined$readClassType) |
+    notCompatibleIds <- (!grepl("compatible", rowDataCombined$readClassType) |
         rowDataCombined$readClassType == "equal:compatible") #keep equal for FDR calculation
-    exonRangesCombined <- exonRangesCombined[notCompatibleIds]
-    rowDataCombined <- rowDataCombined[notCompatibleIds,]
+    subsetTranscripts <- combindRowDataWithRanges(
+        rowDataCombined[!notCompatibleIds,], 
+        exonRangesCombined[!notCompatibleIds])
+    rowDataCombined$maxTxScore[grepl("compatible", rowDataCombined$readClassType) &
+        rowDataCombined$readClassType != "equal:compatible"]=0
   }
   #(2) remove transcripts below NDR threshold/identical junctions to annotations
   rowDataCombined = calculateNDROnTranscripts(rowDataCombined, 
                         useTxScore = length(annotationGrangesList)==0)
-
   if(length(annotationGrangesList)>0){ #only recommend an NDR if its possible to calculate an NDR
       NDR = recommendNDR(rowDataCombined, baselineFDR, NDR, defaultModels, verbose)
   } else {
       if(is.null(NDR)) NDR = 0.5
   }
-  filterSet = (rowDataCombined$NDR <= NDR)
+  filterSet = (rowDataCombined$NDR <= NDR | rowDataCombined$readClassType == "equal:compatible")
```

Comment:
why the equal read class needed here? would they just be the annotated transcripts?